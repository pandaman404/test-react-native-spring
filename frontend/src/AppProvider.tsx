import { NavigationContainer } from '@react-navigation/native';
import { StatusBar } from 'expo-status-bar';
import { ReactNode } from 'react';
import { SafeAreaProvider } from 'react-native-safe-area-context';

interface AppProviderProps {
  children: ReactNode;
}

export default function AppProvider({ children }: AppProviderProps) {
  return (
    <SafeAreaProvider>
      <NavigationContainer>
        <StatusBar style='auto' />
        {children}
      </NavigationContainer>
    </SafeAreaProvider>
  );
}
