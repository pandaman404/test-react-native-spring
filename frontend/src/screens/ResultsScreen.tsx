import { View, Text, StyleSheet, ActivityIndicator } from 'react-native';
import { Colors } from '../constants/colors';
import { useHandleResultsChart } from '../hooks/useHandleResultsChart';
import { SafeAreaView } from 'react-native-safe-area-context';
import GoBackBtn from '../components/GoBackBtn';
import Title from '../components/Title';
import CustomPieChart from '../components/CustomPieChart';
import ErrorMessage from '../components/ErrorMessage';

export default function ResultsScreen() {
  const { chartData, isLoading, errorMessage } = useHandleResultsChart();

  return (
    <SafeAreaView style={styles.container}>
      <GoBackBtn />
      <View style={styles.chartContainer}>
        <Title text='Resultados' />
        {isLoading && <ActivityIndicator size='large' color={Colors.blue} />}
        {!isLoading && chartData.length > 0 && <CustomPieChart pieData={chartData} />}
        {!isLoading && errorMessage && errorMessage.length > 0 && <ErrorMessage text={errorMessage} />}
        {!isLoading && !errorMessage && chartData.length === 0 && (
          <Text style={styles.noResultsText}>No hay resultados</Text>
        )}
      </View>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    gap: 40,
    backgroundColor: Colors.white,
  },
  chartContainer: {
    alignSelf: 'center',
    width: '75%',
    alignItems: 'center',
  },
  noResultsText: {
    color: Colors.black,
    fontSize: 14,
  },
});
