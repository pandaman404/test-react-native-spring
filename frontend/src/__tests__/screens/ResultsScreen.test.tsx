import renderer from 'react-test-renderer';
import AppProvider from '../../AppProvider';
import ResultsScreen from '../../screens/ResultsScreen';

describe('<ResultsScreen />', () => {
  it('Should render ResultsScreen', () => {
    const tree = renderer
      .create(
        <AppProvider>
          <ResultsScreen />
        </AppProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
