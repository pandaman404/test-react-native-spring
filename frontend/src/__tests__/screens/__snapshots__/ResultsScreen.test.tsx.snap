// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultsScreen /> Should render ResultsScreen 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
